File: src/statesman/commands/PrintCombineCommand.java
package statesman.commands;

import statesman.Content;

public class PrintCombineCommand implements Command {

    public static final String ID = "printc";

    protected boolean _initialized;
    protected String[] _messages;

    public PrintCombineCommand() {
        _initialized = false;
        _messages = null;
    }

    public PrintCombineCommand(String[] messages) {
        this();
        _messages = messages;
    }

    @Override
    public Command createInstance(String[] arguments) {
        if (arguments.length > 2) {
            String[] messages = new String[arguments.length - 1];
            System.arraycopy(arguments, 1, messages, 0, arguments.length - 1);
            return new PrintCombineCommand(messages);
        }
        return null;
    }

    @Override
    public void execute() {
        initializeStrings();
        for (int i = 0; i < _messages.length; i++) {
            System.out.printf(_messages[i] + "%n");
        }
    }

    protected void initializeStrings() {
        if (!_initialized) {
            for (int i = 0; i < _messages.length; i++) {
                boolean unescapeString = _messages[i].startsWith("@");
                _messages[i] = Content.getScript().getMessages().getOrDefault(_messages[i], _messages[i]);
                if (unescapeString) {
                    _messages[i] = _messages[i].replace("\\e", "\033");
                }
            }
            _initialized = true;
        }
    }
    
}

File: src/Commands/PrintCombineCommand.cs
using System;

namespace Statesman.Commands
{
    public class PrintCombineCommand : Command
    {
        public static readonly string ID = "printc";

        protected bool _initialized;
        protected string[] _messages;

        public PrintCombineCommand()
        {
            _initialized = false;
            _messages = null;
        }

        public PrintCombineCommand(string[] messages)
                : this()
        {
            _messages = messages;
        }

        public override Command CreateInstance(string[] arguments)
        {
            if (arguments.Length > 2)
            {
                string[] messages = new string[arguments.Length - 1];
                Array.Copy(arguments, 1, messages, 0, arguments.Length - 1);
                return new PrintCombineCommand(messages);
            }
            return null;
        }

        public override void Execute()
        {
            InitializeStrings();
            for (int i = 0; i < _messages.Length; i++)
            {
                Console.Write(_messages[i]);
            }
        }

        protected void InitializeStrings()
        {
            if (!_initialized)
            {
                for (int i = 0; i < _messages.Length; i++)
                {
                    bool unescapeString = _messages[i].StartsWith("@");
                    if (Content.Script.Messages.TryGetValue(_messages[i], out string messageValue))
                    {
                        _messages[i] = messageValue;
                    }
                    if (unescapeString)
                    {
                        _messages[i] = _messages[i].Replace("\\e", "\033");
                    }                }
                _initialized = true;
            }
        }
    }
}
